### PR TITLE
consider hyphen along with space as a potential place to wrap text

### PR DESCRIPTION
**Pull request by nadako**

### CODE REVIEW

File: src/openfl/_internal/text/TextEngine.hx [hunk 1]
```diff
@@ -213,6 +213,16 @@ class TextEngine {
 		}
 	}
 
+	function getWordSeparatorIndex(startIndex:Int):Int {
```
Comment (Neu2o) [inline src/openfl/_internal/text/TextEngine.hx:216]: I would likely unify the `getWordSeparatorIndex` and `getLineBreakIndex` and the way the return if/else statements are written.
You could also unify the default parameter of `startIndex` to twice be `0` or twice being not set?

Comment (nadako) [reply src/openfl/_internal/text/TextEngine.hx:216]: Yep, I also noticed that it would be nice to do this 👍

**Comment by nadako** [comment]
This is not 100% correct, because it removes the separator character, while the hyphen should NOT be removed. So this needs more work.

File: src/openfl/_internal/text/TextEngine.hx [hunk 2]
```diff
@@ -213,16 +213,24 @@ class TextEngine {
...
+		return
+			if (space == -1) hyphen
+			else if (hyphen == -1) space
+			else if (space < hyphen) space
+			else hyphen;
+	}
+
+	public function getLineBreakIndex(startIndex:Int):Int {
 		var cr = text.indexOf("\n", startIndex);
 		var lf = text.indexOf("\r", startIndex);
```
Comment (Neu2o) [inline src/openfl/_internal/text/TextEngine.hx:228]: btw. as I see it now: CR should actually be "\r" and LF should be "\n", but that's quite minor ;)

**Comment by nadako** [comment]
I'll make another PR

**Pull request closed without merging**